src/shared/lib/svgToPdfDownload.ts

```diff
@@ -1,32 +1,53 @@
 import fileDownload from 'react-file-download';
-import request from "superagent";
+import {default as request, Request} from "superagent";
 
 function base64ToArrayBuffer(base64:string) {
-    var binaryString = window.atob(base64);
-    var binaryLen = binaryString.length;
-    var bytes = new Uint8Array(binaryLen);
-    for (var i = 0; i < binaryLen; i++) {
-        var ascii = binaryString.charCodeAt(i);
+    const binaryString = window.atob(base64);
+    const binaryLen = binaryString.length;
+    const bytes = new Uint8Array(binaryLen);
+    for (let i = 0; i < binaryLen; i++) {
+        const ascii = binaryString.charCodeAt(i);
         bytes[i] = ascii;
     }
     return bytes;
 }
 
 export default function (filename:string, svg:Element, servletUrl?: string) {
+    const promise = svgToPdfRequest(svg, servletUrl);
+
+    if (!promise) {
+       return false;
+    }
+
+    promise.end((err, res)=>{
```

Comment:
promise.end.  is that standard?  maybe we use then and catch?

Reply:
i guess this could be specific to what library returns so ...  just wondering

Reply:
@alisman Actually this is not a standard promise. This is a `Request` which extends `Promise`. This code segment was already there, I just needed to make some changes to the existing method to be able to reuse it as a regular promise in the download controls without breaking the previous use-case. I think I should rename the variable here to `request` to make it clear.